libraries/Microsoft.Bot.Builder/Streaming/StreamingRequestHandler.cs

```diff
@@ -371,7 +368,25 @@ private void Server_Disconnected(object sender, DisconnectedEventArgs e)
 
         private async Task ReconnectAsync(IDictionary<string, string> requestHeaders = null)
         {
-            // TODO: Need to get authentication headers from the httpClient on the adapter.
+            // The ServiceUrl of a streaming connection follows the pattern "urn:[ServiceName]:[Protocol]:[ChannelName]".
+            var streamingUrnPattern = new Regex("urn:(.+?:){2}.+");
+            string[] urnSections;
+            try
+            {
+                urnSections = streamingUrnPattern.Split(ServiceUrl);
```

Comment:
Regexes are sensitive to RDDOS attacks. In general its a security best practice to pass a timeout to hte regex.

Reply:
Added time out and case settings on line 372